File: app/src/main/java/com/jsh/kr/alltest/model/JavaScriptManager.java
package com.jsh.kr.alltest.model;

import android.webkit.JavascriptInterface;

public class JavaScriptManager {

   public static final String TAG = JavaScriptManager.class.getSimpleName();
   private OnJavaScriptManagerListener listener;

   public void setJavaScriptManagerListener(OnJavaScriptManagerListener listener) {
      this.listener = listener;
   }

   @JavascriptInterface
   public void onResult(String result) {
      listener.onResult(result);
   }

   @JavascriptInterface
   public void getRandomValue() {
      listener.getRandomValue();
   }

   public interface OnJavaScriptManagerListener {
      void onResult(String result);

      void getRandomValue();
   }
}

File: appkt/src/main/java/com/jsh/kr/alltestkt/model/JavaScriptManager.kt
package com.jsh.kr.alltestkt.model

import android.webkit.JavascriptInterface

class JavaScriptManager {

    private var listener: OnJavaScriptManagerListener? = null

    fun setJavaScriptManagerListener(listener: OnJavaScriptManagerListener?) {
        this.listener = listener
    }

    @JavascriptInterface
    fun onResult(result: String?) {
        listener?.onResult(result)
    }

    @JavascriptInterface
    fun getRandomValue() {
        listener?.getRandomValue()
    }

    interface OnJavaScriptManagerListener {
        fun onResult(result: String?)
        fun getRandomValue()
    }

    companion object {
        val TAG: String = JavaScriptManager::class.java.simpleName
    }
}